tests: make `nvme-cli - nvme_writeuncor_test` optional

The `nvme-cli - nvme_writeuncor_test` should do a feature detection. Currently, it will fail when run against a Qemu nvme device instance:

```
root@localhost:/home/wagi/work/nvme-cli/.build-debian#  /usr/bin/python3 /home/wagi/work/nvme-cli/tests/tap_runner.py --start-dir /home/wagi/work/nvme-cli/tests nvme_writeuncor_test
TAP version 13
1..1
# Using nvme binary 'nvme'
# Running: find /sys/devices -name \*nvme0 | grep -i pci
# /sys/devices/pci0000:00/0000:00:05.0/nvme/nvme0
#
# Running: nvme id-ctrl /dev/nvme0 --output-format=json
# {
#   "vid":6966,
#   "ssvid":6900,
#   "sn":"nvme-0              ",
#   "mn":"QEMU NVMe Ctrl                          ",
#   "fr":"10.1.5  ",
#   "rab":6,
#   "ieee":5395456,
#   "cmic":0,
#   "mdts":7,
#   "cntlid":0,
#   "ver":66560,
#   "rtd3r":0,
#   "rtd3e":0,
#   "oaes":256,
#   "ctratt":32768,
#   "rrls":0,
#   "bpcap":0,
#   "nssl":0,
#   "plsi":0,
#   "cntrltype":1,
#   "fguid":"00000000-0000-0000-0000-000000000000",
#   "crdt1":0,
#   "crdt2":0,
#   "crdt3":0,
#   "crcap":0,
#   "nvmsr":0,
#   "vwci":0,
#   "mec":0,
#   "oacs":298,
#   "acl":3,
#   "aerl":3,
#   "frmw":3,
#   "lpa":7,
#   "elpe":0,
#   "npss":0,
#   "avscc":0,
#   "apsta":0,
#   "wctemp":343,
#   "cctemp":373,
#   "mtfa":0,
#   "hmpre":0,
#   "hmmin":0,
#   "tnvmcap":0,
#   "unvmcap":0,
#   "rpmbs":0,
#   "edstt":0,
#   "dsto":0,
#   "fwug":0,
#   "kas":0,
#   "hctma":0,
#   "mntmt":0,
#   "mxtmt":0,
#   "sanicap":0,
#   "hmminds":0,
#   "hmmaxd":0,
#   "nsetidmax":0,
#   "endgidmax":0,
#   "anatt":0,
#   "anacap":0,
#   "anagrpmax":0,
#   "nanagrpid":0,
#   "pels":0,
#   "domainid":0,
#   "kpioc":0,
#   "mptfawr":0,
#   "megcap":0,
#   "tmpthha":0,
#   "cqt":0,
#   "cdpa":0,
#   "mup":0,
#   "ipmsr":0,
#   "msmt":0,
#   "sqes":102,
#   "cqes":68,
#   "maxcmd":0,
#   "nn":256,
#   "oncs":1885,
#   "fuses":0,
#   "fna":0,
#   "vwc":7,
#   "awun":0,
#   "awupf":0,
#   "icsvscc":0,
#   "nwpc":0,
#   "acwu":0,
#   "ocfs":15,
#   "sgls":524289,
#   "mnan":0,
#   "maxdna":0,
#   "maxcna":0,
#   "oaqd":0,
#   "rhiri":0,
#   "hirt":0,
#   "cmmrtd":0,
#   "nmmrtd":0,
#   "minmrtg":0,
#   "maxmrtg":0,
#   "trattr":0,
#   "mcudmq":0,
#   "mnsudmq":0,
#   "mcmr":0,
#   "nmcmr":0,
#   "mcdqpc":0,
#   "subnqn":"nqn.2019-08.org.qemu:nvme-0",
#   "ioccsz":0,
#   "iorcsz":0,
#   "icdoff":0,
#   "fcatt":0,
#   "msdbd":0,
#   "ofcs":0,
#   "dctype":0,
#   "ccrl":0,
#   "psds":[
#     {
#       "max_power":2500,
#       "max_power_scale":0,
#       "non-operational_state":0,
#       "entry_lat":16,
#       "exit_lat":4,
#       "read_tput":0,
#       "read_lat":0,
#       "write_tput":0,
#       "write_lat":0,
#       "idle_power":0,
#       "idle_scale":0,
#       "active_power":0,
#       "active_power_work":0,
#       "active_scale":0,
#       "emerg_power_fail_recover_time":0,
#       "emerg_power_fail_recover_scale":0,
#       "force_quiesce_vault_time":0,
#       "force_quiesce_vault_scale":0,
#       "emerg_power_fail_vault_time":0,
#       "emerg_power_fail_vault_scale":0
#     }
#   ]
# }
#
# setup: ctrl: /dev/nvme0, ns1: /dev/nvme0n1, default_nsid: 1, flbas: 0
# Running: nvme id-ns /dev/nvme0n1 --output-format=json
# {
#   "nsze":2097152,
#   "ncap":2097152,
#   "nuse":2097152,
#   "nsfeat":52,
#   "nlbaf":7,
#   "flbas":0,
#   "mc":3,
#   "dpc":31,
#   "dps":0,
#   "nmic":0,
#   "rescap":0,
#   "fpi":0,
#   "dlfeat":1,
#   "nawun":0,
#   "nawupf":0,
#   "nacwu":0,
#   "nabsn":0,
#   "nabo":0,
#   "nabspf":0,
#   "noiob":0,
#   "nvmcap":0,
#   "npwg":0,
#   "npwa":0,
#   "npdg":7,
#   "npda":7,
#   "nows":0,
#   "mssrl":0,
#   "mcl":0,
#   "msrc":0,
#   "kpios":0,
#   "nulbaf":0,
#   "kpiodaag":0,
#   "anagrpid":0,
#   "nsattr":0,
#   "nvmsetid":0,
#   "endgid":0,
#   "nguid":"00000000000000000000000000000000",
#   "eui64":"0000000000000000",
#   "lbafs":[
#     {
#       "lbaf":0,
#       "ms":0,
#       "ds":9,
#       "rp":0,
#       "in_use":1
#     },
#     {
#       "lbaf":1,
#       "ms":8,
#       "ds":9,
#       "rp":0,
#       "in_use":0
#     },
#     {
#       "lbaf":2,
#       "ms":16,
#       "ds":9,
#       "rp":0,
#       "in_use":0
#     },
#     {
#       "lbaf":3,
#       "ms":64,
#       "ds":9,
#       "rp":0,
#       "in_use":0
#     },
#     {
#       "lbaf":4,
#       "ms":0,
#       "ds":12,
#       "rp":0,
#       "in_use":0
#     },
#     {
#       "lbaf":5,
#       "ms":8,
#       "ds":12,
#       "rp":0,
#       "in_use":0
#     },
#     {
#       "lbaf":6,
#       "ms":16,
#       "ds":12,
#       "rp":0,
#       "in_use":0
#     },
#     {
#       "lbaf":7,
#       "ms":64,
#       "ds":12,
#       "rp":0,
#       "in_use":0
#     }
#   ],
#   "vs":[
#   ]
# }
#
# Running: nvme read /dev/nvme0n1 --start-block=1023 --block-count=0 --data-size=512 --data=nvmetests/TestNVMeUncor/read_file.txt
# read: Success
#
# Running: nvme write-uncor /dev/nvme0n1 --start-block=1023 --block-count=0
# NVMe status: Invalid Command Opcode: A reserved coded value or an unsupported value in the command opcode field(0x4001)
#
not ok 1 - test_write_uncor (TestNVMeUncor)
  ---
  traceback: |
    Traceback (most recent call last):
      File "/usr/lib/python3.11/unittest/case.py", line 57, in testPartExecutor
        yield
      File "/usr/lib/python3.11/unittest/case.py", line 623, in run
        self._callTestMethod(testMethod)
      File "/usr/lib/python3.11/unittest/case.py", line 579, in _callTestMethod
        if method() is not None:
           ^^^^^^^^
      File "/home/wagi/work/nvme-cli/tests/nvme_writeuncor_test.py", line 74, in test_write_uncor
        self.assertEqual(self.write_uncor(), 0)
    AssertionError: 1 != 0
  ...
```

--- a/tests/nvme_writeuncor_test.py
+++ b/tests/nvme_writeuncor_test.py
@@ -30,6 +30,7 @@ NVMe Write Compare Testcae:-
 
 """
 
+from nvme_test import to_decimal
 from nvme_test_io import TestNVMeIO
 
 
@@ -51,6 +52,9 @@ class TestNVMeUncor(TestNVMeIO):
         self.read_file = self.test_log_dir + "/" + self.read_file
         self.create_data_file(self.write_file, self.data_size, "15")
         open(self.read_file, 'a').close()
+        oncs = to_decimal(self.get_id_ctrl_field_value("oncs"))
+        if not oncs & (1 << 1):
+            self.skipTest("write-uncor is not supported by this controller")
 
     def tearDown(self):
         """ Post Section for TestNVMeUncor """

